packages/flutter/lib/src/cupertino/colors.dart

```diff
@@ -539,45 +537,106 @@ class CupertinoColors {
-// [DefaultTextStyle]), the component will automatically resolve the color by calling
-// [CupertinoDynamicColor.resolve], using their own [BuildContext].
-//
-// When used outside of a Cupertino component, such color resolution will not happen
-// automatically. It's essential to call [CupertinoDynamicColor.resolve] with the
-// correct [BuildContext] before using the color to paint, in order to get the
-// desired effect.
+/// [CupertinoDynamicColor.color] will be the default effective color.
+///
+/// Sometimes manually resolving a [CupertinoDynamicColor] is not necessary, because
+/// the Cupertino Library provides built-in support for it.
+///
+/// ### Using a [CupertinoDynamicColor] in a Cupertino widget
+///
+/// When a Cupertino widget is provided with a [CupertinoDynamicColor], either
+/// directly in its constructor, or from an [InheritedWidget] it depends on (for example,
+/// [DefaultTextStyle]), the widget will automatically resolve the color using
+/// [CupertinoDynamicColor.resolve] against its own [BuildContext], on a best-effort
+/// basis.
+///
+/// {@tool sample}
+/// The following snippet builds a [CupertinoButton] that appears white in light
+/// mode, and changes automatically to black in dark mode.
+///
+/// ```dart
+/// Builder(builder: (BuildContext context) {
+///   return CupertinoButton(
+///     child: child,
+///     // CupertinoDynamicColor works out of box in a CupertinoButton.
+///     color: CupertinoDynamicColor.withBrightness(
+///       color: CupertinoColors.white,
+///       darkColor: CupertinoColors.black,
+///     ),
+///     onPressed: () => null,
+///   );
+/// })
+/// ```
+/// {@end-tool}
+///
+/// ### Using a [CupertinoDynamicColor] from a [CupertinoTheme]
+///
+/// When referring to a [CupertinoTheme] color, generally the color will already
+/// have adapted to the ambient [BuildContext], because [CupertinoTheme.of]
+/// implicitly resolves all the colors used in the retrieved [CupertinoThemeData],
+/// before returning it.
+///
+/// {@tool sample}
+/// The following snippet creates a [Container] with the `primaryColor` of the
+/// current theme. If `primaryColor` is a [CupertinoDynamicColor], the container
+/// will be adaptive, thanks to [CupertinoTheme.of]: it will switch to `primaryColor`'s
+/// dark variant once dark mode is turned on, and turns to primaryColor`'s high
+/// contrast variant when [MediaQueryData.highContrast] is requested in the ambient
+/// [MediaQuery], etc.
+///
+/// ```dart
+/// Builder(builder:(BuildContext context) {
+///   // Container is not a Cupertino widget, but CupertinoTheme.of implicitly resolves
+///   // colors used in the retrieved CupertinoThemeData.
+///   return Container(
+///     color: CupertinoTheme.of(context).primaryColor,
+///   );
+/// })
+/// ```
+/// {@end-tool}
+///
+/// ### Manually Resolving a [CupertinoDynamicColor]
+///
+/// When used outside of a Cupertino component, or wrapped in an object opaque to
```

Comment:
Changed to
> When used to configure a non-Cupertino widget, or wrapped in an object opaque   
to the receiving Cupertino component, a [CupertinoDynamicColor] may need to be  
manually resolved using [CupertinoDynamicColor.resolve], before it can used     
to paint. For example, to use a custom [Border] in a [CupertinoNavigationBar],  
the colors used in the [Border] have to be resolved manually before being passed
to [CupertinoNavigationBar]'s constructor.